### PR TITLE
Improve menu popup focus when opened

### LINKED ISSUE #1900: Shift focus when menu's popup is initially opened
# Feature Request

## Description
When the collapsible menu view's popup is initially open, it may make sense to shift focus to first focusable element within it.

### LINKED ISSUE #1919: VoiceOver on mobile focus is not shifted to popup in tabs when opened.
# Bug Report

## Description
 Focus is not shifted to tabs in popup when popup is opened. User is not able to easily maneuver into popup when opened.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-ui/#/components/terra-tabs/tabs/tabs
2. Click on button to open tabs popup.

## Expected Behavior
VO focus shifts to first tab item in popup.

## Possible Solution
Look into shifting focus to first tab item in popup when opened from compact tabs view.

## Environment

* Component Name and Version: terra-popup
* Browser Name and Version: Mobile Safari 12 on iOS 12 w/ VoiceOver

### LINKED ISSUE #2015: Menu stack off by 1 when opening submenu with enter or space key
# Bug Report

## Description
I noticed this issue when testing via keyboard. Opening the submenu by focusing on the menu item that triggers open the submenu and pressing the `Enter` or `Space` key causes the menu stack to increase by 2 instead of 1.

If you click on the menu item that triggers open the submenu or when focused on the menu item that triggers open the submenu and press right arrow key, only 1 item is added to the array `stack`.

This results in additional keyboard events needed to close the submenu. The user has to hit enter or space on the back button twice to reduce that stack to its initial state.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-core/#/raw/tests/terra-menu/menu/menu/sub-menu
1. Click on the button on the page
1. Open the react dev tools and select the menu component in the DOM so you can view it's state. The `stack` array in state should have just 1 item in it.
1. Press `tab` key, then press `down arrow` key
1. On this step, press either `Enter` or `Space` key. 
1. 2 items are added to the array `stack` totaling `3` items in the `stack`.

To close the submenu with enter, you have to press `tab` and then `enter` and then `tab` and then `enter` again to reduce the `stack` back down to 1 item.

Video of steps from above.
[menu-stack-issue.mp4.zip](https://github.com/cerner/terra-core/files/2539951/menu-stack-issue.mp4.zip)

## Expected Behavior
Pressing enter or space on the menu item that triggers open the submenu should only increase the stack by one.

## Possible Solution
Look into event handlers and see how we are getting this error and refactor it.

## Environment

* Component Name and Version: terra-menu, terra-collapsible-menu-view
* Browser Name and Version: Tested in Mobile Safari 12 w/ VoiceOver on iOS and Chrome
* Operating System and version (desktop or mobile): Mac, iOS 12

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Resolves #1900 
Resolves #1919 
Resolves #2015 

### Deployment Links
https://terra-core-deployed-pr-2004.herokuapp.com/#/components/terra-menu/menu/menu

https://terra-core-deployed-pr-2004.herokuapp.com/#/components/terra-collapsible-menu-view/collapsible-menu-view/collapsible-menu-view

https://terra-core-deployed-pr-2004.herokuapp.com/#/components/terra-tabs/tabs/tabs

### CODE REVIEW

File: packages/terra-menu/src/_MenuContent.jsx [hunk 1]
```diff
@@ -135,8 +135,8 @@ class MenuContent extends React.Component {
 
   validateFocus(node) {
     if (this.needsFocus && node) {
-      node.focus();
-      this.needsFocus = document.activeElement !== node;
```
Comment (bjankord) [inline packages/terra-menu/src/_MenuContent.jsx]: This shifts focus to the first menu item in the menu when opened, (this first interactable element in the menu).

File: packages/terra-menu/src/Menu.module.scss [hunk 2]
```diff
@@ -43,6 +43,10 @@
       word-wrap: normal;
     }
 
+    .header-button-container {
+      outline: 0;
```
[the remark this answers is not in the document]
Comment (bjankord) [reply packages/terra-menu/src/Menu.module.scss]: Ended up removing this.

**Comment by bjankord** [comment]
Talked with @mmalaker about adding up down arrow key support on menu when you first open it, shifting focus to first / last item in it.

We also talked about disabling right arrow shifting to the first menu item in the tabs menu when it is opened via enter/spacebar. We want to treat keyboard interaction between the tabs (left / right arrow keys) to cycle between tabs and the tabs menu (up / down arrow keys) as separate so if a user knows how to navigate a menu already, the expected keys don't change when it's part of a tab component.

**Comment by bjankord** [comment]
@yuderekyu Yeah, I don't have a good fix for #2014. Plan to leave that out of scope for this PR.

**Comment by mmalaker** [comment]
During screen reader testing I found issues with each of the components. It seems there is some conflict between keyboard navigation and screen reader actions when those actions share the same keys.
- For menus, when you get to the nested menu, the screen reader will leave the menu and start reading the underlying page. Left and right arrow keys do not work for navigation between menu and submenus when using JAWS.
- For collapsible menu view, when you enter the menu using VoiceOver, and attempt to navigate using VoiceOver commands, it says "Safari busy" and will not navigate through the list items. You can access the menu using regular keyboard nav, but focus will move off of the menu and back onto the main page while the menu is still open.
- For tabs, using the arrow keys to navigate between tabs doesn't work reliably. Sometimes you can't tab to or arrow back to the previous tabs. Also, JAWS does not read the tab text when switching between tabs in a group using the arrow keys.

**Comment by bjankord** [comment]
@mmalaker I've logged issues for the items identified in functional verification. In regards to the changes on this PR, where you able to verify the following issues were resolved:

* https://github.com/cerner/terra-core/issues/1900
* https://github.com/cerner/terra-core/issues/1919

**Comment by mmalaker** [comment]
Functional verification completed. The menu popup examples have the correct focus behavior. Other screen reader issues to be addressed separately.

**Pull request merged**